File: 1.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var terms = <int, int>{};
    for (int i = 0; i < nums.length; i++) {
      if (terms.containsKey(target - nums[i])) {
        return [terms[target - nums[i]]!, i];
      }
      terms[nums[i]] = i;
    }
    throw Exception();
  }
}

File: 1.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut num_index: HashMap<i32, i32> = HashMap::new();
        for (i, num) in nums.iter().enumerate() {
            if let Some(&index) = num_index.get(&(target - num)) {
                return [i as i32, index].to_vec();
            }
            num_index.insert(*num, i as i32);
        }
        unreachable!()
    }
}
